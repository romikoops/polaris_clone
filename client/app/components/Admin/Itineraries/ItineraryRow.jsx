import React, { PureComponent } from 'react'
import { v4 } from 'uuid'
import styles from './index.scss'
import PropTypes from '../../../prop-types'
import { switchIcon, gradientTextGenerator } from '../../../helpers'

class ItineraryRow extends PureComponent {
  viewSchedules () {
    const { adminDispatch, itinerary } = this.props
    adminDispatch.loadItinerarySchedules(itinerary.id, true)
  }
  viewPricings () {
    const { adminDispatch, itinerary } = this.props
    adminDispatch.getItineraryPricings(itinerary.id, true)
  }

  render () {
    const {
      theme,
      itinerary
    } = this.props

    const [startCity, endCity] = itinerary.name.split(' - ')
    const gradientIcon = {
      ...gradientTextGenerator(theme.colors.primary, theme.colors.secondary)
    }

    return (
      <div className={`flex-100 layout-row layout-align-start-start ${styles.itinerary_row}`} key={v4()} >
        <div className={`flex-15 layout-row layout-align-center-center height_100 ${styles.mot_icon}`}>
          {switchIcon(itinerary.mode_of_transport, gradientIcon, 'flex-none')}
        </div>

        <div className={`flex-45 layout-row layout-align-space-around-center ${styles.itinerary_row_box}`}>
          <div className={`flex-none ${styles.divider}`} />
          <div className={`flex layout-row layout-wrap layout-align-center-center ${styles.itinerary_row_city}`}>
            <p className={`flex-100 ${styles.city_name}`}>{startCity}</p>
            <p className={`flex-100 ${styles.city_context}`}>Departure</p>
          </div>
          <div className={`flex-15 layout-row layout-align-center-center ${styles.chevron_icon}`}>
            <i
              className="flex-none fa fa-angle-double-right clip"
              style={gradientIcon}
            />
          </div>
          <div className={`flex layout-row layout-wrap layout-align-center-center ${styles.itinerary_row_city}`}>
            <p className={`flex-100 ${styles.city_name}`}>{endCity}</p>
            <p className={`flex-100 ${styles.city_context}`}>Arrival;</p>
          </div>
        </div>
        <div className={`flex layout-row layout-align-center-center pointy ${styles.itinerary_row_box} ${styles.itinerary_row_btn}`} onClick={() => this.viewSchedules()} >
          <div className={`flex-none ${styles.divider}`} />
          <p className="flex-none">View Schedules</p>
        </div>

        <div className={`flex layout-row layout-align-center-center pointy ${styles.itinerary_row_box} ${styles.itinerary_row_btn}`} onClick={() => this.viewPricings()}>
          <div className={`flex-none ${styles.divider}`} />
          <p className="flex-none">View Pricings</p>
        </div>
      </div>
    )
  }
}

ItineraryRow.propTypes = {
  theme: PropTypes.theme,
  itinerary: PropTypes.objectOf(PropTypes.any).isRequired,
  adminDispatch: PropTypes.objectOf(PropTypes.func).isRequired
}
ItineraryRow.defaultProps = {
  theme: {}
}
export default ItineraryRow
