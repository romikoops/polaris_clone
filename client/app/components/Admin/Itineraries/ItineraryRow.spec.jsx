import * as React from 'react'
import { shallow } from 'enzyme'
import { theme } from '../../../mock'

import ItineraryRow from './ItineraryRow'

jest.mock('uuid', () => {
  let counter = -1
  const v4 = () => {
    counter += 1

    return `RANDOM_KEY_${counter}`
  }

  return { v4 }
})

const itineraryBase = {
  name: 'FOO - BAR',
  mode_of_transport: 'air'
}

const propsBase = {
  itinerary: itineraryBase,
  adminDispatch: {},
  theme
}

test('shallow render', () => {
  expect(shallow(<ItineraryRow {...propsBase} />)).toMatchSnapshot()
})
