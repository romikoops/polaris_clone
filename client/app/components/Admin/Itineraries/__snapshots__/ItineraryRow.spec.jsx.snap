// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow render 1`] = `
<div
  className="flex-100 layout-row layout-align-start-start itinerary_row"
  key="RANDOM_KEY_0"
>
  <div
    className="flex-15 layout-row layout-align-center-center height_100 mot_icon"
  >
    <i
      className="fa fa-plane clip flex-none"
      style={
        Object {
          "background": "-webkit-linear-gradient(-90deg, #333,#fafafa)",
        }
      }
    />
  </div>
  <div
    className="flex-45 layout-row layout-align-space-around-center itinerary_row_box"
  >
    <div
      className="flex-none divider"
    />
    <div
      className="flex layout-row layout-wrap layout-align-center-center itinerary_row_city"
    >
      <p
        className="flex-100 city_name"
      >
        FOO
      </p>
      <p
        className="flex-100 city_context"
      >
        Departure
      </p>
    </div>
    <div
      className="flex-15 layout-row layout-align-center-center chevron_icon"
    >
      <i
        className="flex-none fa fa-angle-double-right clip"
        style={
          Object {
            "background": "-webkit-linear-gradient(-90deg, #333,#fafafa)",
          }
        }
      />
    </div>
    <div
      className="flex layout-row layout-wrap layout-align-center-center itinerary_row_city"
    >
      <p
        className="flex-100 city_name"
      >
        BAR
      </p>
      <p
        className="flex-100 city_context"
      >
        Arrival;
      </p>
    </div>
  </div>
  <div
    className="flex layout-row layout-align-center-center pointy itinerary_row_box itinerary_row_btn"
    onClick={[Function]}
  >
    <div
      className="flex-none divider"
    />
    <p
      className="flex-none"
    >
      View Schedules
    </p>
  </div>
  <div
    className="flex layout-row layout-align-center-center pointy itinerary_row_box itinerary_row_btn"
    onClick={[Function]}
  >
    <div
      className="flex-none divider"
    />
    <p
      className="flex-none"
    >
      View Pricings
    </p>
  </div>
</div>
`;
